File: backend/hospital-service/src/main/java/com/bharatteeka/hospital/entity/Slot.java
package com.bharatteeka.hospital.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Table(name = "slot")
@Data
public class Slot {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "slot_id")
    private Integer slotId;

    @Column(name = "slot_date")
    private LocalDate date;

    @Column(name = "start_time")
    private LocalTime startTime;

    @Column(name = "end_time")
    private LocalTime endTime;

    @Column(name = "capacity")
    private Integer capacity;

    @Column(name = "booked_count")
    private Integer bookedCount;

    @ManyToOne
    @JoinColumn(name = "vaccine_id", nullable = false)
    private Vaccine vaccine;  // renamed from vaccineId → vaccine

    @Column(name = "is_booked")
    private Boolean isBooked;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    // Convenience method to calculate available slots
    public Integer getAvailable() {
        if (capacity == null || bookedCount == null) return 0;
        return capacity - bookedCount;
    }
    
    @Column(name = "is_active",nullable = false)
    private Boolean isActive = true;

}

File: backend/AdminService/AdminService/Models/Slot.cs
using System;
using System.Collections.Generic;

namespace AdminService.Models;

public partial class Slot
{
    public int SlotId { get; set; }

    public int HospitalId { get; set; }

    public DateOnly SlotDate { get; set; }

    public TimeOnly StartTime { get; set; }

    public TimeOnly EndTime { get; set; }

    public int Capacity { get; set; }

    public int? BookedCount { get; set; }

    public int VaccineId { get; set; }

    public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();

    public virtual Hospital Hospital { get; set; } = null!;

    public virtual ICollection<Notification> Notifications { get; set; } = new List<Notification>();

    public virtual Vaccine Vaccine { get; set; } = null!;
}
